sdmetrics/single_table/privacy/disclosure_protection.py

```diff
@@ -243,3 +297,232 @@ def compute(
             num_discrete_bins,
         )
         return score_breakdown['score']
+
+
+class DisclosureProtectionEstimate(DisclosureProtection):
+    """DisclosureProtectionEstimate metric."""
+
+    @classmethod
+    def _validate_inputs(
+        cls,
+        real_data,
+        synthetic_data,
+        known_column_names,
+        sensitive_column_names,
+        computation_method,
+        continuous_column_names,
+        num_discrete_bins,
+        num_rows_subsample,
+        num_iterations,
+    ):
+        super()._validate_inputs(
+            real_data,
+            synthetic_data,
+            known_column_names,
+            sensitive_column_names,
+            computation_method,
+            continuous_column_names,
+            num_discrete_bins,
+        )
+        if not isinstance(num_rows_subsample, int) or num_rows_subsample <= 0:
+            raise ValueError('`num_rows_subsample` must be an integer greater than zero.')
+
+        if not isinstance(num_iterations, int) or num_iterations <= 0:
+            raise ValueError('`num_iterations` must be an integer greater than zero.')
+
+    @classmethod
+    def _compute_estimated_cap_metric(
+        cls,
+        real_data,
+        synthetic_data,
+        baseline_protection,
+        known_column_names,
+        sensitive_column_names,
+        computation_method,
+        num_rows_subsample,
+        num_iterations,
+        verbose,
+    ):
+        estimation_iterator = tqdm.tqdm(range(num_iterations), disable=(not verbose))
+        if verbose:
+            description = 'Estimating Disclosure Protection (Score={score:.3f})'
+            estimation_iterator.set_description(description.format(score=0))
+
+        cap_metric = CAP_METHODS.get(computation_method)
+        estimated_score_sum = 0
+        for i in estimation_iterator:
+            real_data_samp = real_data.sample(min(num_rows_subsample, len(real_data)))
+            synth_data_samp = synthetic_data.sample(min(num_rows_subsample, len(synthetic_data)))
```

Comment:
what happens if the length of these two arrays differ because one has less than the number of rows to subsample and the other doesn't? Does that matter?